trl/models/utils.py

```diff
@@ -150,7 +150,7 @@ def unwrap_model_for_generation(
     if accelerator.state.deepspeed_plugin is not None and accelerator.state.deepspeed_plugin.zero_stage == 3:
         with deepspeed.zero.GatheredParameters(model.parameters()):
             remove_hooks(model)
-            yield model
+            yield accelerator.unwrap_model(model)
```

Comment:
I believe models wrapped with the `DeepSpeedEngine` can still generate, so I'm curious why this is needed

Reply:
Ah it causes issues for RLOO, having some errors like "DeepSpeedEngine has not attribute generate`, so we still need to unwrap it.